gateway/tests/tests/tests.rs

```diff
@@ -129,20 +102,21 @@ async fn test_gateway_authentication() -> Result<()> {
         amount: bitcoin::Amount::from_sat(100),
         address: bitcoin.get_new_address(),
     };
+    test_auth(&gw_password, |pw| client_ref.withdraw(pw, payload.clone())).await?;
+
+    task_group.shutdown_join_all().await
+}
+
+async fn test_auth<Fut>(gw_password: &str, func: impl Fn(String) -> Fut) -> Result<()>
```

Comment:
Docstrings please! :D

"Test that a given endpoint/functionality fails with the wrong password but works with the correct one?" :)

Reply:
:sweat_smile: